src/components/Dropdowns/SelectionDropdown.tsx

```diff
@@ -195,14 +183,28 @@ export default function SelectionDropdown(inputProps: SelectionDropdownProps): R
     [props.showSelectedItemTooltip, props.controlTooltipPlacement]
   );
 
-  // Create an ID for the HTML label element if one is provided.
-  const id = props.label ? `dropdown-label-${props.label.toLowerCase().replaceAll(" ", "_")}` : undefined;
+  // Create an ID for the HTML label element if one is provided.;
+  useEffect(() => {
+    if (!props.label && !props.id) {
+      console.log(
+        "SelectionDropdown: No label or id provided for the dropdown, which means that the select component may not be labeled correctly for screen readers." +
+          " Consider either providing the `label` prop, or setting the `id` prop and passing it an HTML `label` via the `for` attribute."
+      );
+    }
+  }, []);
+  const selectId = props.id ?? "selection-dropdown-" + props.label?.toLowerCase().replaceAll(" ", "_");
+  const labelId = props.label ? selectId + "-label" : undefined;
 
   return (
     <FlexRowAlignCenter $gap={6}>
-      {props.label && <h3 id={id ?? props.htmlLabelId}>{props.label}</h3>}
+      {props.label && (
+        <label htmlFor={selectId}>
+          <h3 id={labelId}>{props.label}</h3>
+        </label>
+      )}
       <StyledSelect
-        aria-labelledby={id}
```

Comment:
I was previously using `aria-labelledby` to try and connect the select input with the label, but from what I understand, `aria` doesn't interact with the HTML `label` system. `aria` is metadata that can help screen readers, but using `label` for form elements has direct functional implications.

Reply:
You can test this change by clicking on the label text of any of the dropdowns. This should now cause the dropdown to be focused.

<img width="453" height="90" alt="Image" src="https://github.com/user-attachments/assets/fc53f404-005c-4cc7-842d-532d9b904c99" />